src/domain/usecases/CopyDataSetsOUUserCase.ts

```diff
@@ -0,0 +1,96 @@
+import _ from "lodash";
+import { DataSetsRepository } from "domain/repositories/DataSetsRepository";
+import { PostOptions, Ref } from "types/d2-api";
+import { DataSet } from "domain/entities/DataSet";
+import { OUCopyResult } from "domain/entities/OUCopyResult";
+
+export class CopyDataSetsOUUserCase {
+    constructor(
+        private dataSetsRepository: DataSetsRepository
+    ) { }
+
+    async execute(options: {
+        originDataset: string;
+        destinationDatasets: string[];
+        replace: boolean | undefined;
+    }): Promise<OUCopyResult> {
+        const { originDataset, destinationDatasets, replace } = options;
+
+        console.debug(`Replace the destination OU: ${replace ? "True" : "False"}`);
+
+        const datasets = await this.dataSetsRepository.get([originDataset, ...destinationDatasets]);
+
+        const origDataset = datasets[originDataset];
+        const destDatasets = destinationDatasets.map((id) => datasets[id]);
+
+        if (!origDataset || !destDatasets) throw new Error("Missing DataSets");
```

Comment:
Empty JS arrays are always truthy (unlike Python), so this condition will be never met. You can use `_.compact` and check that the resulting array has the expected length, for example. `if (!destDataSet` will be no longer necessary.

Reply:
That if its not for checking if they are empty, but to deal with this:
TS2345: Argument of type 'DataSet | undefined' is not assignable to parameter of type 'DataSet'.
  Type 'undefined' is not assignable to type 'DataSet'.
datasets[originDataset] always will exists because this.dataSetsRepository.get will fail previously otherwise.

Reply:
Sorry, it was confusing, I was referring to connected, but different things. I meant that if you use a compact on the array, then you remove undefined, and later the check is not necessary. But the way you've refactored it (keep T | undefined and check inside the map) is perfectly fine.

Reply:
Used `_.compact` to lessen the check clutter.